### ISSUE TITLE
JS Not Minified

### ISSUE THREAD

**Issue by codeclinic** [holger1411/understrap-child#87]
Gulp doesn't appear to be creating a minified version of child-theme.js, although it minifies the css files.

From what I could understand of the gulp file, there is no task for minifying js files. Is this intended? Can js minification be added to the watch process?

**Comment by codeclinic** [comment]
After a bit of a search through other issues, i see that the child-theme.js file isnt intended to be added to or edited in anyway. New scripts need to be added in a new .js file.

**Comment by codeclinic** [comment]
I see the js folder is being watched by gulp, but I don't understand what happens after a new js file is added. Should this be converted to a minified version or combined with another file?

**Comment by axlright** [comment]
You can modify the gulpfile.js to add watch directories for additional .js files. This is what I've done:

`gulp.task('scripts', function() {
    var scripts = [
        basePaths.dev + 'js/tether.js', // Must be loaded before BS4

        // Start - All BS4 stuff
        basePaths.dev + 'js/bootstrap4/bootstrap.js',
        // End - All BS4 stuff

        // Start - All Custom stuff
        basePaths.dev + 'js/custom/*.js',
        // End - All Custom stuff

        basePaths.dev + 'js/skip-link-focus-fix.js'
    ];`

Then I add custom .js files to /src/custom/ directory and then run gulp scripts to compile the js. Works well.

-Alex

**Comment by fadingdust** [comment]
@axlright Right now, I'm seeing the ./src directory in the gitignore file. Did you change that too? 

Traditionally, I've committed the code in ./src/js & ./src/sass for the gulp-minify sources. But if I add ./src/js to be watched, it gives an errors thanks to core.js & customizer.js being included unnecessarily. 

@holger1411 I'd like some clarification on the intention of the ./src and ./dist folders, specifically, where does my per-theme JS go? I see the ./sass folder for per-theme css, but no equivalent for js. 

This seems like a sensible layout for gulpfile.js. 

// Defining base paths
var basePaths = {
    node: './node_modules/', //vendor sources: gitignore
    vendor: './vendor/' //copy here from node_modules via gulp copy-assets: gitignore
    dev: './src/' //local sass & js to be minified: commit
    js: './js/', // final compiled files included in theme: optionally commit
};

**Comment by axlright** [comment]
Yeah now I have my custom scripts in the directory above /src/ directory, also because the /src/ directory gets re-written on updates.

I now have custom scripts in /custom-scripts/ and this is my gulpfile:

```
gulp.task('scripts', function() {
    var scripts = [

        // Start - All BS4 stuff
        basePaths.dev + 'js/bootstrap4/bootstrap.js',
        // End - All BS4 stuff

        // Start - All Custom stuff
        basePaths.dev + '../custom-scripts/*.js',
        // End - All Custom stuff

        basePaths.dev + 'js/skip-link-focus-fix.js'
    ];
```

**Comment by fadingdust** [comment]
Indeed the least intrusive way seems to be to just add a ./scripts folder, like the ./sass folder, but that organization doesn't seem to denote "these are what get transpiled." 

I think what's bugging me is that none of these seem to follow any organizational consistency rules:
<pre>
./img/src --> ./img
./sass ---> ./css  // this is used by Automattic's Underscores :/
./????  ---> ./js
</pre>
 
Also, I think I might have some confusion on what @holger1411 uses the gulpfile for (./dist?) & what the end-user (like us) would use it for. Not sure what the best answer is for separating those 2 roles out (copy-assets goes into ./vendor instead of ./src is my thought), but I'd at least like to see the transpilation/minification folder situation cleaned up, though Automattic isn't helping my cause :D

@codeclinic I'm probably stealing your thread :/ Thoughts?
